plugins/amazonq/mynah-ui/src/mynah-ui/ui/quickActions/handler.ts

```diff
@@ -47,19 +47,59 @@ export class QuickActionHandler {
                 this.handleHelpCommand(tabID)
                 break
             case '/transform':
-                this.handleGumbyCommand(tabID)
+                this.handleCodeTransformCommand(tabID, eventId)
                 break
             case '/clear':
                 this.handleClearCommand(tabID)
                 break
         }
     }
 
-    private handleGumbyCommand(tabID: string) {
-        if (!this.isGumbyEnabled) {
+    private handleCodeTransformCommand(tabID: string, eventId?: string) {
+        if (!this.isCodeTransformEnabled) {
             return
         }
-        this.connector.transform(tabID)
+
+        // Check for existing opened transform tab
+        const existingTransformTab = this.tabsStorage.getTabs().find((tab) => tab.type === 'codetransform')
+        if (existingTransformTab !== undefined) {
+            this.mynahUI.selectTab(existingTransformTab.id, eventId || "")
+            this.connector.onTabChange(existingTransformTab.id)
+
+            this.mynahUI.notify({
+                title: "Code Transformation",
+                content: "Switched to opened transformation tab",
+            });
+            return
+        }
+
+        // Add new tab
+        let affectedTabId: string | undefined = tabID
+        if (this.tabsStorage.getTab(affectedTabId)?.type !== 'unknown') {
+            affectedTabId = this.mynahUI.updateStore('', {})
+        }
+        if (affectedTabId === undefined) {
+            this.mynahUI.notify({
+                content: uiComponentsTexts.noMoreTabsTooltip,
+                type: NotificationType.WARNING,
+            })
+            return
+        } else {
+            this.tabsStorage.updateTabTypeFromUnknown(affectedTabId, 'codetransform')
+            this.connector.onKnownTabOpen(affectedTabId)
+            // Clear unknown tab type's welcome message
+            this.mynahUI.updateStore(affectedTabId, {chatItems: []})
+            this.mynahUI.updateStore(affectedTabId, this.tabDataGenerator.getTabData('codetransform', true))
+            this.mynahUI.updateStore(affectedTabId, {
+                promptInputDisabledState: true,
+                promptInputPlaceholder: 'Chat disabled during Code Transformation.',
```

Comment:
This should be localized

Reply:
I.e. I guess put in `uiComponentsTexts`? Not sure you have access to `MessagesBundle.properties` from ts code.

Reply:
I think .ts file text do not access MessagesBundle.properties file. Can we get a confirmation @rli ?

Reply:
Resolved. See -> https://github.com/aws/aws-toolkit-jetbrains/pull/4187/files#r1529019906